z3/src/lib.rs

```diff
@@ -175,3 +175,10 @@ pub struct Params<'ctx> {
     ctx: &'ctx Context,
     z3_params: Z3_params,
 }
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum LiftedBool {
```

Comment:
After renaming this to `SATResult` or something ... please also add a doc comment here explaining what it is for.